code/game/machinery/camera/camera_assembly.dm

```diff
@@ -31,6 +31,10 @@
 	if(building)
 		setDir(ndir)
 
+/obj/structure/camera_assembly/Destroy()
+	upgrades = null

```

Comment:
I find it weird that the prox_sensor is getting qdel'd but not the fucking assembly